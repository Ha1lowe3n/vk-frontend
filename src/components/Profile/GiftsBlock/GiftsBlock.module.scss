.giftsBlock {
    margin-top: 15px;

    .gifts {
        padding: 12px 15px 12px;
        cursor: pointer;
        width: inherit;
        display: flex;
        justify-content: space-between;

        img {
            height: 64px;
            width: 64px;
        }
    }
}
