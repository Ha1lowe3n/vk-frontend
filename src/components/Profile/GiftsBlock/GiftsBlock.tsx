import React from "react";
import { WhiteBlock } from "../../WhiteBlock/WhiteBlock";
import { WhiteBlockInfo } from "../../WhiteBlock/WhiteBlockInfo/WhiteBlockInfo";

import styles from "./GiftsBlock.module.scss";

export const GiftsBlock: React.FC = () => {
    return (
        <WhiteBlock className={styles.giftsBlock}>
            <WhiteBlockInfo title={"Gifts"} count={105} />
            <div className={styles.gifts}>
                <img src="https://vk.com/sticker/4-1360-96" alt="Team Spirit" />
                <img src="https://vk.com/sticker/4-540-96" alt="VK Music" />
                <img src="https://vk.com/images/gift/1143/96.png" alt="Cake" />
            </div>
        </WhiteBlock>
    );
};
