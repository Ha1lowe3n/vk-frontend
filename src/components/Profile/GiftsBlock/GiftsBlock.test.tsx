import { render, screen } from "@testing-library/react";
import { GiftsBlock } from "./GiftsBlock";

it("render GiftsBlock component", () => {
    render(<GiftsBlock />);
    expect(screen.getByText("Gifts")).toBeInTheDocument();
    expect(screen.getByText("105")).toBeInTheDocument();
});

it("GiftsBlock snapshot", () => {
    const view = render(<GiftsBlock />);
    expect(view).toMatchSnapshot();
});
