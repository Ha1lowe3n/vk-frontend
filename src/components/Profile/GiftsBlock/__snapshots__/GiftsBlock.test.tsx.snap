// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiftsBlock snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="whiteBlock giftsBlock"
      >
        <div
          class="whiteBlockInfo"
        >
          <span
            class="title"
          >
            Gifts
          </span>
          <span
            class="count"
          >
            105
          </span>
        </div>
        <div
          class="gifts"
        >
          <img
            alt="Team Spirit"
            src="https://vk.com/sticker/4-1360-96"
          />
          <img
            alt="VK Music"
            src="https://vk.com/sticker/4-540-96"
          />
          <img
            alt="Cake"
            src="https://vk.com/images/gift/1143/96.png"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="whiteBlock giftsBlock"
    >
      <div
        class="whiteBlockInfo"
      >
        <span
          class="title"
        >
          Gifts
        </span>
        <span
          class="count"
        >
          105
        </span>
      </div>
      <div
        class="gifts"
      >
        <img
          alt="Team Spirit"
          src="https://vk.com/sticker/4-1360-96"
        />
        <img
          alt="VK Music"
          src="https://vk.com/sticker/4-540-96"
        />
        <img
          alt="Cake"
          src="https://vk.com/images/gift/1143/96.png"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
